CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# Contributing
+
+We love pull requests from everyone. By participating in this project, you
+agree to abide by the Mozilla [code of conduct][coc].
+
+Developer setup steps are covered in the [README][dev setup]. This file explains how our
+dev process works. These are just guidelines; nothing here is intended to be
+rigid or confusing. If you'd like to contribute and don't understand something
+here, reach out on IRC, and we'll be happy to offer solutions.
+
+[coc]: https://www.mozilla.org/en-US/about/governance/policies/participation
+[dev setup]: https://github.com/mozilla/universal-search-addon#developer-setup
+
+### Saying Hello
+
+If you're new to the project, please say hello in IRC! We can help you succeed
+by finding issues that are a good fit for your skills/interests, and ensuring
+you have a contact person to guide you through the bug fixing process, answer
+any questions, and help if you get stuck.
+
+Find us in #universal-search on Mozilla IRC. If you need help with IRC setup,
+the Mozilla [wiki][wiki] has instructions.
+
+[wiki]: https://wiki.mozilla.org/IRC
+
+### Contributing code
+
+We use the same process for community members and paid staff.
+
+#### Branches
+
+Fork the project and create one branch per bug on your personal repo.
+
+The branch name should describe the bug, not the fix. It should start with the bug number, and include a very short description of the bug, or a short version of
+the bug's title, like `150-broken-image-data`.
+
+#### Commit messages
+
+Commit messages should generally follow [this helpful advice][commit], but please
+start your commit message with "Fixes #nn:", which eases skimming history later.
```

Comment:
Not sure I agree that "Fixes #nn" helps skimmability, since bug numbers don't have any meaning themselves. I've generally been a fan of:

> Capitalized, short (50 chars or less) summary (fixes #NN).